composer/utils/ddp.py

```diff
@@ -141,6 +141,49 @@ def all_reduce(
                        "The mosaic trainer will automatically do this for you.")
 
 
+def broadcast(tensor: torch.Tensor, src: int = 0) -> None:
+    """Broadcasts the tensor to the whole group.
+
+    ``tensor`` must have the same number of elements in all processes participating in the collective.
+    See :meth:`torch.distributed.broadcast`.
+
+    Args:
+        tensor (torch.Tensor): Data to be sent if ``src`` is the rank of current process,
+            and tensor to be used to save received data otherwise.
+        src (int, optional): Source rank (default: ``0``)
+    """
+    if dist.is_available() and dist.is_initialized():
+        dist.broadcast(tensor, src)
+    world_size = get_world_size()
+    if world_size == 1:
+        return
+    raise RuntimeError(f"Since the world_size({world_size}) > 1, please configure DDP to use ddp.broadcast(). "
```

Comment:
Shouldn't this error be raised at the start of the function?

Reply:
It's only an issue if there is no ddp and there's more than one node, so the check is down here.